runtime/battery-station/src/parameters.rs

```diff
@@ -155,31 +155,35 @@ parameter_types! {
     pub const DisputeFactor: Balance = 2 * BASE;
     /// Maximum Categories a prediciton market can have (excluding base asset).
     pub const MaxCategories: u16 = MAX_CATEGORIES;
+    /// Maximum block period for a dispute.
+    pub const MaxDisputeDuration: BlockNumber = MAX_DISPUTE_DURATION;
     /// Maximum number of disputes.
     pub const MaxDisputes: u16 = 6;
-    /// Minimum number of categories. The trivial minimum is 2, which represents a binary market.
-    pub const MinCategories: u16 = 2;
-    // 60_000 = 1 minute. Should be raised to something more reasonable in the future.
-    /// Minimum number of milliseconds a Rikiddo market must be in subsidy gathering phase.
-    pub const MinSubsidyPeriod: Moment = 60_000;
+    /// Maximum string length for edit reason.
+    pub const MaxEditReasonLen: u32 = 1024;
+    /// Maximum block period for a grace_period.
+    /// The grace_period is a delay between the point where the market closes and the point where the oracle may report.
+    pub const MaxGracePeriod: BlockNumber = MAX_GRACE_PERIOD;
+    pub const MaxMarketLifetimeInBlocks: BlockNumber = MAX_MARKET_LIFETIME_IN_BLOCKS;
+    pub const MaxMarketLifetimeInMoments: Moment = MAX_MARKET_LIFETIME_IN_MOMENTS;
```

Comment:
Docstring missing.

Reply:
Why are we double-booking the docs of the pallet config parameters for every runtime that we maintain? I wouldn't do that for reasons of abstractions. Right now, when the doc changes, we need to change it in three different places.

Reply:
Side-by-side comparison with and without docstrings. Which one's easier to read, edit and alphabetize? And let's be real here for a second: Does `MaxMarketPeriod` really need the docstring "The maximum market period"?

<img width="1920" alt="Screenshot 2022-12-22 at 11 57 54" src="https://user-images.githubusercontent.com/49552287/209119868-31f7262c-6441-4e39-8526-15cce283bd40.png">

Reply:
> Why are we double-booking the docs of the pallet config parameters for every runtime that we maintain? I wouldn't do that for reasons of abstractions. Right now, when the doc changes, we need to change it in three different places.

We've done that because those are three separate crates that deserve their own documentation and most of the parameters are present in the runtime crates, but not in the primitives crate. When you inspect the documentation of each of the crates, you can see an appropriate documentation. 

<details>
  <summary>
  Battery Station Documentation
  </summary>
  <img src=https://user-images.githubusercontent.com/15378334/209358656-bc18c014-9db2-4118-974e-cff1d29b475d.png></img>
</details


If we remove them from the runtime crates, most of the types will be undocumented. For the others that are present in the primitives crate and documented: How should one know to look there? I think for us this is obvious, for new developers it probably won't be so straightforward to figure out what these values are for.

I see the problem of having to update the docs in three places if they change.
I am not really open to remove all parameter docs from the runtime crates, because most of the documentation is not available in the parameters crate, however I would be open to leave the docs away that are present in primitives and add a top-level docstring to the parameters module in the runtime crates that references to the constant module in the primitives crate for the missing documentation.

Reply:
> Side-by-side comparison with and without docstrings. Which one's easier to read, edit and alphabetize? And let's be real here for a second: Does `MaxMarketPeriod` really need the docstring "The maximum market period"?
> 
> <img alt="Screenshot 2022-12-22 at 11 57 54" width="1920" src="https://user-images.githubusercontent.com/49552287/209119868-31f7262c-6441-4e39-8526-15cce283bd40.png">

I think this is not an issue of the docstrings (in fact that help devs to know what it's about), but of the syntax highlighting. In the right example we see the docstrings being highlighted in a way that puts them in the foreground. At the same time the the constants are also highlighted in a way that puts them in the foreground (just not as much as the docstrings, so the docstrings are visually more conspicuous). I bet that if the docstrings in the right example would be highlighted like the comments in the left example (gray, which renders them inconspicuous relative to the dark gray background), they would not bother us in the right example and the actual code would be perceived more continuous.

In regards to `MaxMarketPeriod`, disregarding actual implementation details, I agree that the names are self-explanatory. I am currently questioning if we should include documentation about implementation details, for example: When exactly is the market period counted? Is it from market creation to resolution? Market start to market close? Market creation to market close? It would help a bit when setting this value, however, it would introduce double booking again: If the implementation details in regards to `MaxMarketPeriod` in PM change, we also have to update the docs in runtime. This is error-prone. What is your opinion about this?
I remember when creating the [XCM configuration and structures](https://github.com/zeitgeistpm/zeitgeist/blob/75bcae6e21c15b6fbf91e3f8aee76adfe7f6bf9d/runtime/zeitgeist/src/xcm_config/config.rs) needed for that, I decided to go with docstrings that include implementation details when useful in this context because there are a lot of different components that are dispersed all over different crates and it takes quite some time to figure out what is going on. That way we have it all in one place.

Reply:
> I think this is not an issue of the docstrings (in fact that help devs to know what it's about), but of the syntax highlighting. In the right example we see the docstrings being highlighted in a way that puts them in the foreground. At the same time the the constants are also highlighted in a way that puts them in the foreground (just not as much as the docstrings, so the docstrings are visually more conspicuous). I bet that if the docstrings in the right example would be highlighted like the comments in the left example (gray, which renders them inconspicuous relative to the dark gray background), they would not bother us in the right example and the actual code would be perceived more continuous.

That's true, grey would work much better here.

> I see the problem of having to update the docs in three places if they change.
> I am not really open to remove all parameter docs from the runtime crates, because most of the documentation is not available in the parameters crate, however I would be open to leave the docs away that are present in primitives and add a top-level docstring to the parameters module in the runtime crates that references to the constant module in the primitives crate for the missing documentation.

I understand. I'm okay with that. As long as it's not just a copy-paste situation, I'm okay with doubling the docs. We're not changing them often.

> If the implementation details in regards to MaxMarketPeriod in PM change, we also have to update the docs in runtime. This is error-prone. What is your opinion about this?

I think some terms (like _market lifetime_. and everything assocaited with the lifecycle of a markete) should rather be defined in a "meta documentation". Otherwise you end up not only double-booking between the runtimes and the pallets, but also between different objects in `zrml_prediction_markets`. For example, the market life cycle needs to be explained in one place. Otherwise, the description of the market cycle is split between `ReportingPeriod`, `MinOracleDuration`, `MaxOracleDuration`, etc. Once you understand how the state transitions work, you don't require long explanations in the docstrings. The documentation page is a good place for that, I think. Not sure if that really answer the question, though.

Reply:
[85cf8ce](https://github.com/zeitgeistpm/zeitgeist/pull/908/commits/85cf8ce82c98c06632adbdb0474bd1640ddd9b15)